contrib/cirrus/runner.sh

```diff
@@ -55,31 +57,45 @@ function _run_unit() {
 }
 
 function _run_apiv2() {
+    _bail_if_test_can_be_skipped test/apiv2
```

Comment:
Does this say only test APIv2 if there were changes in the test/apiv2 directory?

Reply:
No, it's the opposite.  Ed fixed the comment that handles the arguments:

```
# If PR touches any files in an argument directory, we cannot skip
```

Reply:
(and the code does what the comment says :grin:)

Reply:
But I dont think we should be skipping apiv2 tests if the test/apiv2 directory was not changed.  If the pkg/api directory or anything in pkg was changed, it could break the apiv2 tests.

Reply:
right, I follow you now, yes this was confusing for me too.  So the subsequent block removes all 'test/*' and 'docs/*' changes, then only blocks testing if there are no other changes.  So, if in your example some podman code was changed, the apiv2 tests **would** run.  It would only be skipped if apiv2 changes *only* included other changes under `test/*` and/or `docs/*`.  It's confusing, but I think Ed's got it right.

Reply:
```
    # If we still have diffs, they indicate files outside of docs & test.
    # It is not safe to skip.
    if [[ -n "$diffs" ]]; then
        return 0
    fi
```

Reply:
@rhatdan let's see if this clarifies: `_bail_if_test_can_be_skipped TEST/FOO` will bail if TEST/FOO can be skipped. A test can be skipped if:
1. the PR does not modify any files under TEST/FOO; and
2. the PR does not modify any files other than in `docs` and `test`.

That is: if PR touches only files in `test/e2e`, we can skip `test/system`.

If PR touches any file outside of `docs` and `test`, any file at all whatsoever, the full CI suite will run. Does that help?

Reply:
Got it now I understand.